lesson10/src/task4/HeavyBox.java

```diff
@@ -0,0 +1,31 @@
+package task4;
+
+
+public class HeavyBox implements Comparable<HeavyBox> {
+    private int id;
+    private int weigh;
+
+
+    public HeavyBox(int id, int weigh) {
+        this.id = id;
+        this.weigh = weigh;
+
+    }
+
+
+    @Override
+    public String toString() {
+        return "HeavyBox{" +
+                " id= " + id +
+                ", weigh= " + weigh +
+                '}';
+    }
+
+
+    @Override
+    public int compareTo(HeavyBox o) {
+        if (this.weigh <= o.weigh) {
```

Comment:
обычно когда равны возвращается 0.
```suggestion
        return this.weigh - o.weigh;
```